dask_ml/cluster/spectral.py

```diff
@@ -216,7 +220,9 @@ def fit(self, X, y=None):
         # compute the exact blocks
         # these are done in parallel for dask arrays
         if isinstance(X, da.Array):
+            logger.info("Starting small rechunk-persist")
             X_keep = X[keep].rechunk(self.n_components).persist()
```

Comment:
Note that rechunk takes in a chunk size, not a desired number of chunks

Reply:
Yes, in this case, `n_components` is small. We want this to be a single chunk w/ `n_components` chunksize. IIRC, I needed this to help with performance in the pairwise_kernels later on.

Reply:
OK, the name here just threw me off is all.  I was afraid that you might have mistaken the intent of the argument.  You might want at least `.rechunk({0: self.n_components})` to ensure that you don't accidentally rechunk the second axis down to chunks of size `n_components` as well (presumably this would only be an issue if you had very many columns)